assets/src/domain/eventEditor/ui/ticketAssignmentsManager/data/useAssignmentManager.ts

```diff
@@ -19,108 +20,141 @@ const useAssignmentManager = (): AM => {
-			relation: 'datetimes',
-			relationId: datetimeId,
-		};
+	const toggleAssignment = useCallback<AM['addAssignment']>(
+		({ datetimeId, ticketId }) => {
+			const assignedTickets = getAssignedTickets({ datetimeId });
+			const remove = assignedTickets.includes(ticketId);
+			updateAssignment({ datetimeId, ticketId, remove });
+		},
+		[getAssignedTickets, updateAssignment]
+	);
 
-		if (remove) {
-			removeRelation(datetimeToTickets);
-			removeRelation(ticketsToDatetimes);
-		} else {
-			// Add both ways relation for fast retieval
-			addRelation(datetimeToTickets);
-			addRelation(ticketsToDatetimes);
-		}
-	};
+	const removeAssignment = useCallback<AM['removeAssignment']>(
+		({ datetimeId, ticketId }) => {
+			updateAssignment({ datetimeId, ticketId, remove: true });
+		},
+		[updateAssignment]
+	);
 
-	const initialize: AM['initialize'] = ({ data, assignmentType, entity }) => {
-		const relationsToPick: Array<keyof TAMRelationalData> = ['datetimes', 'tickets'];
-		// pick only datetimes and tickets from relational data
-		let newData = pick(relationsToPick, data);
+	const initialize = useCallback<AM['initialize']>(
+		({ data, assignmentType, entity }) => {
+			const relationsToPick: Array<keyof TAMRelationalData> = ['datetimes', 'tickets'];
+			// pick only datetimes and tickets from relational data
+			let newData = pick(relationsToPick, data);
 
-		// Remove other relations from newData
-		// like ticket to price relations
-		newData = mapObjIndexed((relationalEntity, entityType) => {
-			let relationalEntityToUse = relationalEntity;
-			// If TAM is only for a single datetime/ticket
-			// limit relations to that datetime/ticket
-			if (
-				(assignmentType === 'forDate' && entityType === 'datetimes') ||
-				(assignmentType === 'forTicket' && entityType === 'tickets')
-			) {
-				relationalEntityToUse = pick([entity.id], relationalEntity);
-				// if it's for a new date or ticket
-				if (isEmpty(relationalEntityToUse)) {
-					const newRelationKey = entityType === 'datetimes' ? 'tickets' : 'datetimes';
-					// init to empty relations
-					relationalEntityToUse[entity.id] = {
-						[newRelationKey]: [],
-					};
+			// Remove other relations from newData
+			// like ticket to price relations
+			newData = mapObjIndexed((relationalEntity, entityType) => {
+				let relationalEntityToUse = relationalEntity;
+				// If TAM is only for a single datetime/ticket
+				// limit relations to that datetime/ticket
+				if (
+					(assignmentType === 'forDate' && entityType === 'datetimes') ||
+					(assignmentType === 'forTicket' && entityType === 'tickets')
+				) {
+					relationalEntityToUse = pick([entity.id], relationalEntity);
+					// if it's for a new date or ticket
+					if (isEmpty(relationalEntityToUse)) {
+						const newRelationKey = entityType === 'datetimes' ? 'tickets' : 'datetimes';
+						// init to empty relations
+						relationalEntityToUse[entity.id] = {
+							[newRelationKey]: [],
+						};
+					}
```

Comment:
maybe extract for the sake of readability... 
would need to be curried so that `relationsToPick` can be used ti initialize the function

Reply:
Extracted, thanks